### PR TITLE
Improvements from wiper hackathon

### PR DESCRIPTION
**Pull request by BjoernAtBosch**
# Description

Add improvements from the wiper hackathon

## Azure DevOps PBI/Task reference

AB#_11305_

## Checklist

* [ ] Release workflow is passing

### CODE REVIEW

**Comment by BjoernAtBosch** [comment]
Hi Dominic, hi Markus,

was that all to take over?

File: sdk/include/sdk/AsyncResult.h [hunk 1]
```diff
@@ -163,6 +163,17 @@ template <typename TResultType> class AsyncResult {
      */
     [[nodiscard]] bool isInAwaitingState() const { return m_awaiting; }
 
+    template <typename TType>
```
Comment (doosuu) [inline sdk/include/sdk/AsyncResult.h:166]: Doxygen missing.

**Review by doosuu — CHANGES_REQUESTED** [review]

File: sdk/include/sdk/AsyncResult.h (hunk 1 above)
Comment (BjoernAtBosch) [reply sdk/include/sdk/AsyncResult.h:166]: @doosuu: Could you add some - I'm not exactly aware what this function is doing?

Comment (doosuu) [reply sdk/include/sdk/AsyncResult.h]: @BjoernAtBosch added commit with doxygen documentation added.

**Review by doosuu — APPROVED** [review]

**Pull request merged**